appender.go

```diff
@@ -240,6 +256,14 @@ func (a *Appender) Add(ctx context.Context, index string, document io.Reader) er
 	return nil
 }
 
+func (a *Appender) addProcessedCount(delta int64, lm *int64, m metric.Int64Counter, status ProcessedStatus) {
+	// legacy metric
+	atomic.AddInt64(lm, delta)
+
+	attr := a.processedEventAttrSet[status]
+	m.Add(context.Background(), delta, metric.WithAttributeSet(attr))
+}
```

Comment:
As above, let's keep it simple and not create a reusable attribute set for the `status` attribute. Instead I think we should remove `addProcessedCount`, and add a `... metric.AddOption` parameter to `addCount`. Then we can call it like this:

`a.addCount(int64(n), &a.tooManyRequests, a.metrics.docsIndexed, metric.WithAttributes(attribute.String("status", "TooMany")))`

WDYT?

Reply:
Thanks for the suggestion. I went with this approach and decided not to do that as we discussed that we wanted to avoid the overhead of creating an attribute set every time. But I like this approach more if the overhead is not a big concern here. I've addressed it in the later commit.